### PR TITLE
Refactored and updated

### PR DESCRIPTION
**Pull request by oliverfoster**
1. Refactored - into: adapt session states, lms session states, offline storage
2. Updated - pipwerks api
3. Added scorm_test_harness.html
    - When run from scorm_test_harness the LMS session variables are stored in a local cookie, allows better debugging when writing components with offline requirements.
4. Updated main.html to a fully responsive version (with no LMS warnings).

See https://github.com/adaptlearning/adapt_framework/pull/571 for added Adapt.offlineStorage interface

### CODE REVIEW

**Comment by belal-almassri-spongeuk** [comment]
:+1:

**Comment by zenduo** [comment]
+1

File: js/scorm/wrapper.js [hunk 1]
```diff
@@ -100,6 +106,11 @@ define (function(require) {
 	ScormWrapper.prototype.initialize = function() {
 		this.lmsConnected = this.scorm.init();
 
+		// the pipwerks code will set the course to 'incomplete' automatically, but it doesn't commit that change
+		if(this.scorm.handleCompletionStatus && this.commitOnStatusChange && this.isFirstSession()) {
+			this.commit();
+		}
+
```
Comment (moloko) [inline js/scorm/wrapper.js]: this if() statement can now be removed as the updated pipwerks code does now call commit when the setting the course to incomplete

**Comment by moloko** [comment]
+1

**Comment by oliverfoster** [comment]
@belal-almassri-spongeuk @zenduo  main.html added back in on request from sponge and kineo, allows for main.html use in production with no tracking or warnings.

**Comment by belal-almassri-spongeuk** [comment]
+1 

Thanks a lot Ollie, this has been tested and works perfectly.

**Comment by dancgray** [comment]
+1

**Comment by zenduo** [comment]
+1 (like it a lot) works on scorm cloud too.

**Pull request merged**